api/api.lisp

```diff
@@ -218,6 +218,13 @@
                 (macro-expand (macro-expand-macro expr ram))
                 (nest-apps (mapcar #'macro-expand expr))))))))))
 
+(defun lurk-eq (a b)
+  (or
+   (eq a b)
+   (and (typep a 'string)
+        (typep b 'string)
+        (equal a b))))
```

Comment:
Do we need the complexity, or can we just define LURK-EQ to be CL:EQUAL? I don't think there are any EQUAL non-strings which are not meant to be LURK-EQ. Lurk equality should be both structural and physical, because of hash-consing.